pilot/pkg/security/authn/v1alpha1/policy_applier.go

```diff
@@ -293,6 +295,29 @@ func convertPolicyToAuthNFilterConfig(policy *authn_v1alpha1.Policy, proxyType m
 	return filterConfig
 }
 
+// setDownstreamTLSContext sets the downstream TLS context configuration based on the information passed to it
+func setDownstreamTLSContext(tls *auth.DownstreamTlsContext, tlsServerRootCert, tlsServerCertChain, tlsServerKey string, requireClientCerts *wrappers.BoolValue, alpnProtocols, subjectAltNames []string) {
+	tls.CommonTlsContext = &auth.CommonTlsContext{
+		AlpnProtocols: alpnProtocols,
+		TlsCertificates: []*auth.TlsCertificate{
```

Comment:
See my comments - since we want to stop using file mounts, it would be better to genereate a SDS block, with a different name for the secret. 

And use the SecretCache to load from the new files. 

I'm not sure you need to pass the full path name to Pilot - something like ISTIO_META_CERT_TYPE or CERT_NAME would work better (shorter and more clear). The node agent/pilot agent can take care of finding the right path name. 

I would also suggest using a default ( consistent with what we do for regular certs ) - for example /var/run/secrets/dnscerts. This would reduce the need for the env variable - you just need to mount the secret at the right location, and pilot agent can detect it. It would also make maintenance easier - people troubleshooting will know where to look instead of digging trough configs for each customer.